Tests/SwiftParserTest/DeclarationTests.swift

```diff
@@ -1371,6 +1371,20 @@ final class DeclarationTests: XCTestCase {
       )
     )
   }
+
+  func testMissingExpressionInVariableAssignment() {
+    AssertParse(
+      """
+      let a =1️⃣
+      """,
+      diagnostics: [
+        DiagnosticSpec(message: "expected expression in variable")
```

Comment:
This is actually harder to implement than I thought because `InitializerClause` is used for
- `FunctionParameter` (as default value)
- `PaternBinding` (this case)
- `EnumCaseElement` (as raw value)
- `MacroDecl`(the macro definition)

And currently the `in` clause just looks at node’s `name_for_diagnostics`. I tried changing it to also look at the missing node child’s `name_for_diagnostic` but that regressed diagnostics in other cases.

So, I’d prefer to keep it as-is for now. We can always try to improve it later if we want but the current diagnostics doesn’t seem tooooo bad to me.